File: io/SortedList.py
class SortedList:
    
    def __init__(self):
        self.list = []
    
    def __str__(self):
        result = ""
        for s in self.list:
            result += s + "\n"
        return result
    
    def add_list(self, names):
        for s in names:
            self.list.append(s)
        self.list.sort()

if __name__ == "__main__":
    list1 = ["Adam", "Sam", "Mike"]
    list2 = ["Becky", "Chelsea", "Molly"]
    list3 = ["Connor", "John", "Scott"]

    sorted_list = SortedList()
    sorted_list.add_list(list1)
    print("After first add:")
    print(sorted_list)

    sorted_list.add_list(list2)
    print("After second add:")
    print(sorted_list)

    sorted_list.add_list(list3)
    print("After third add:")
    print(sorted_list)

File: io/SortedList.java
import java.io.*;
import java.util.*;
 
public class SortedList {
	
	ArrayList<String> list;
	
	public SortedList(){
		
		list = new ArrayList<String>();
	}
	
	public String toString()
	{
		String result = "";
		
		for (String s : list)
		{
			result = result + s + "\n";
		}
		
		return result;
	}
	
	public void addList(String[] names) { 
		
		for (String s : names)
			list.add(s);
		
		Collections.sort(list);
	}
	
	public static void main(String[] args) {

		String[] list1 = {"Adam", "Sam", "Mike"};
		String[] list2 = {"Becky", "Chelsea", "Molly"};
		String[] list3 = {"Connor", "John", "Scott"};

		SortedList list = new SortedList();
		list.addList(list1);
    	System.out.println("After first add:");
    	System.out.println(list.toString());

		list.addList(list2);
    	System.out.println("After second add:");
    	System.out.println(list.toString());

		list.addList(list3);
    	System.out.println("After third add:");
    	System.out.println(list.toString());
	}
}
	
// Sample Output
/*
After first addList:
Adam
Mike
Sam

After second addList:
Adam
Becky
Chelsea
Mike
Molly
Sam

After third addList:
Adam
Becky
Chelsea
Connor
John
Mike
Molly
Sam
Scott  
*/
